### ISSUE TITLE
compileCommands uses only "command" element and ignores "arguments"

### ISSUE THREAD

**Issue by ytimenkov** [microsoft/vscode-cpptools#1890]
I happen to use waf build system with gcc (6.3) and I found that built-in support generates `compile_commands.json` file which looks like this (I've omitted irrelevant flags)
```json
  {
    "directory": "/home/default/project/build/debug", 
    "arguments": [
      "/opt/rh/devtoolset-6/root/usr/bin/g++", 
      "-std=c++14", 
      "-I../../src", 
      "-I../../../.conan/data/boost/1.66.0/external/stable/package/9a0f10048143a0afca8737aa0ed65fb2fdf4d10f/include", 
      "-DPOSIX", 
      "-D_GNU_SOURCE", 
      "-D__STDC_LIMIT_MACROS", 
      "-D__STDC_FORMAT_MACROS", 
      "-DDEBUG", 
      "-DBOOST_USE_STATIC_LIBS", 
      "../../src/folder/file.cpp", 
      "-c", 
      "-o/home/default/project/build/debug/src/folder/file.cpp.1.o"
    ], 
    "file": "../../src/folder/file.cpp"
  }, 
```

When I open file.cpp I see that the extension correctly found compiler and standard include paths but completely ignored explicit defines and include paths.

It's confirmed by log output (set `C_Cpp.loggingLevel` to "6") - no mention of custom includes.

Then reading #1718 I saw that it has slightly different format, namely uses `command` instead of `arguments`, so I converted to single-line and it worked.

According to documentation:
> **arguments:** The compile command executed as list of strings. Either **arguments** or **command** is required.

**Comment by bobbrow** [comment]
This is a bug.  I'm looking at the code and I see that we have support for arguments when the switch is separated from the value, but not when they are merged into the same argument.  For example `"-I", "../../src"`.  Should be a quick fix.

**Comment by ytimenkov** [comment]
I found that it was changed in Waf from"command" to "arguments" only 2 months ago and was easy to roll back (separating `-I` from the path is a more fundamental change).

I wonder if these compile commands are too tied to clang as one needs to replicate parsing logic, not to mention that there could be some flags which affect defines or libs, like -pthread.

**Comment by sean-mcmanus** [comment]
This should be fixed with 0.17.0-insiders2 or later: https://github.com/Microsoft/vscode-cpptools/releases . The final release is planned for early next week.

**Comment by bobbrow** [comment]
This is fixed in 0.17.0.
